src/LaunchDarkly.ServerSdk/Internal/DataModelDependencies.cs

```diff
@@ -1,12 +1,13 @@
-﻿using System.Collections.Generic;
+﻿using System;
+using System.Collections.Generic;
 using System.Linq;
 using LaunchDarkly.Sdk.Server.Internal.Model;
 
 using static LaunchDarkly.Sdk.Server.Interfaces.DataStoreTypes;
 
 namespace LaunchDarkly.Sdk.Server.Internal.DataSources
 {
-    internal struct KindAndKey
+    internal readonly struct KindAndKey : IEquatable<KindAndKey>
```

Comment:
Used as a dictionary and hashset key.  Ran a comparison with 10,000 hashes:

10,000 hashes

Without Fix:
```
5182 ms
5173 ms
5270 ms
5261 ms
5244 ms
5287 ms
5177 ms
5350 ms
5190 ms
5163 ms
```

With Fix:
```
3 ms
0 ms
0 ms
0 ms
1 ms
0 ms
0 ms
0 ms
0 ms
1 ms
```

Originally I tried with 1,000,000 million hashes but got tired of waiting for the old implementation to finish.

Reply:
What this looks like in production right now when we change a bunch of features together:

![image](https://user-images.githubusercontent.com/12101120/152512375-30872283-14fc-4e37-a319-a50d1eeec0cc.png)